### ISSUE TITLE
Integrated only becomes Discrete only after waking from sleep

### ISSUE THREAD

**Issue by taralx** [codykrieger/gfxCardStatus#326]
Steps to reproduce:

1. Set Integrated Only
2. Run Chrome and open something that would activate discrete GPU
3. Close laptop
4. Wait a bit.
5. Open laptop

Expected result:
Integrated GPU in use

Actual result:
Discrete GPU in use, checkmark moved to Discrete only

**Comment by czitzmann** [comment]
Similar issue on my MacBook Pro (15" Late 2011) with Intel HD Graphics 3000 512MB running OS 10.13.6 and gfxCardStatus v.2.3; however, Open apps were Firefox, TeamViewer, Messages, Calendar, Mail, Keychain Access and no sleep required. 

Despite numerous times switching to "integrated only," gfxCardStatus repeatedly (automatically) switched to either "Discrete only" or "Dynamic switching" modes without any restart or user intervention.

Chris

**Comment by codykrieger** [comment]
@czitzmann That's more likely to be #107. This issue is specific to waking from sleep.

**Comment by danparsons** [comment]
I'm having this exact same issue too (on a MacBook Pro (16-inch, 2019) with Intel UHD 630 / Radeon Pro 5500M) on macOS 10.15.6 and the #107 linked above was (1) locked; (2) 5 years ago. Does this mean the bug will never be fixed?

**Comment by codykrieger** [comment]
@danparsons If you're having _this_ issue (i.e. the GPU switches when you wake from sleep), then I think #107 is totally different.

Regardless, as far as bug fixes in general are concerned…gfxCardStatus was last updated in 2012. I don't even have a dual-GPU machine anymore, which really complicates developing and testing fixes. Unless a particularly motivated developer wants to spend a large quantity of time spelunking through the disassembly of AppleGraphicsControl-and-friends and send me a Pull Request, I don't expect this to be fixed anytime soon.

**Comment by danparsons** [comment]
@codykrieger Fair enough, I totally understand. I'm not sure if I can make it happen yet but would a 1-3 month loan of a recent Macbook Pro with dual GPUs assist you in bug fixes?

**Comment by codykrieger** [comment]
@danparsons That's a generous offer. I'm not sure I can commit to anything in the very short term because I'm a week away from starting a new job, and I've just taken on a bunch of contract work to boot, but if that is something you think you could arrange, I'm sure that would be helpful.

**Comment by codykrieger** [comment]
@danparsons Also, not to stray too far off topic—that's a gorgeous guitar you've got in your profile pic. I've been a Les Paul guy forever, but I've recently been bitten by the PRS bug.
